src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1055,6 +1057,8 @@ - (void)_handleSubscriptionError:(NSError *)error
 {
     std::lock_guard lock(_lock);
 
+    MATTER_LOG_METRIC_END(kMetricMTRDeviceSubscriptionSetup, [MTRError errorToCHIPErrorCode:error]);
```

Comment:
So this function is called in three cases:

1. Getting a CASE session for the initial subscription fails.
2. Kicking off the subscription on the CASE session we got fails.
3. We failed to kick off a resubscribe for some reason.

Item 3 does not seem relevant to us, at first glance.  Maybe this should be inside `_handleSubscriptionReset`, though that can also get called if everything here is being torn down.....

On the other hand, how do we want to handle the case of "got a CASE session, but first attempt to set up a subscription failed, so we will try again"?  Right now we will keep waiting for it and eventually when it's established log a "success" end metric.  Is that what we want?

Also: we seem to be treating subscription reattempts that come from MTRDevice (i.e. "before we get a CASE session") and the ones that come from inside ReadClient (i.e. "after we get a CASE session") differently: the former log begin/end on every attempt, while the latter logs begin on first attempt start and then end on success, which might be a while later).  This asymmetry is a bit strange to me.